src/Catalog/Icons/IconCopyResultCache.cs

```diff
@@ -75,6 +78,10 @@ public ExternalIconCopyResult Get(Uri iconUrl)
 
             if (_externalIconCopyResults.TryGetValue(iconUrl, out var result))
             {
+                if (!result.IsCopySucceeded && (!result.Expiration.HasValue || result.Expiration.Value < DateTimeOffset.UtcNow))
+                {
```

Comment:
Minor suggestion: you could reduce nesting by inverting the `if` on line 79. I don't feel strongly about this though. Example:

```csharp
if (!_externalIconCopyResults.TryGetValue(iconUrl, out var result))
{
  return null;
}

if (!result.IsCopySucceeded && ...)
{
  return null;
}

return result;
```